changelogs/fragments/81504-update-logical-operation.yml

```diff
@@ -1,2 +1,2 @@
 bugfixes:
-  - collection dependency resolver - previous code had the same operand on both sides of a logical expression. Fix now updates the operand in the right side of the logical expression (https://github.com/ansible/ansible/pull/81511).
+    - collection dependency resolver - modified the '__lt__' comparison to depend on '_priority' field only.
```

Comment:
Sorry this got lost in the backlog of PRs. It previously only depended on the `_priority` field too, because `self.name < self.name` is False. I don't think it needs a changelog since it's not a change in behavior (a commit message is enough). Do you mind deleting it, or I can push an update to your branch.